datafusion/sql/src/unparser/plan.rs

```diff
@@ -729,12 +722,16 @@ impl Unparser<'_> {
                     .map(|input| self.select_to_sql_expr(input, query))
                     .collect::<Result<Vec<_>>>()?;
 
-                let union_expr = SetExpr::SetOperation {
-                    op: ast::SetOperator::Union,
-                    set_quantifier: ast::SetQuantifier::All,
-                    left: Box::new(input_exprs[0].clone()),
-                    right: Box::new(input_exprs[1].clone()),
-                };
+                let union_expr = input_exprs
```

Comment:
```suggestion
                let Some(union_expr) = input_exprs
```
Instead of unwrapping it directly, returning an error if the option is none is better. You can refer to how L736 did it.